tests/functional/common.js

```diff
@@ -0,0 +1,126 @@
+/* @flow */
+import {ChildProcess, spawn} from 'child_process';
+import copyDir from 'copy-dir';
+import path from 'path';
+import promisify from 'es6-promisify';
+
+
+import * as tmpDirUtils from '../../src/util/temp-dir';
+export const withTempDir = tmpDirUtils.withTempDir;
+
+
+export const functionalTestsDir = path.resolve(__dirname);
+export const projectDir = path.join(functionalTestsDir, '..', '..');
+export const webExt = path.join(projectDir, 'bin', 'web-ext');
+export const fixturesDir = path.join(functionalTestsDir, '..', 'fixtures');
+export const addonPath = path.join(fixturesDir, 'minimal-web-ext');
+export const fakeFirefoxPath = path.join(
+  functionalTestsDir, 'fake-firefox-binary.js'
+);
+export const fakeServerPath = path.join(
+  functionalTestsDir, 'fake-amo-server.js'
+);
+
+
+// withTempAddonDir helper
+
+export type TempAddonParams = {
+  addonPath: string,
+  runFromCwd?: boolean
+};
+
+export type TempAddonCallback =
+  (tmpAddonDir: string, tmpDir: string) => Promise<any>
+
+const copyDirAsPromised = promisify(copyDir);
+export function withTempAddonDir(
+  {addonPath, runFromCwd}: TempAddonParams,
+  makePromise: TempAddonCallback,
+): Promise<any> {
+  return withTempDir((tmpDir) => {
+    const tempAddonDir = path.join(tmpDir.path(), 'tmp-addon-dir');
+    return copyDirAsPromised(addonPath, tempAddonDir)
+      .then(() => {
+        // NOTE: when "web-ext run" is spawned in functional tests,
```

Comment:
d'oh. That's a weird quirky feature of yargs. We had to override it in `package.json`; see https://github.com/mozilla/web-ext/commit/f93d68ab6a64ee3a322d5964c707a5a3cd36d287 . I guess that's why it stops working after you change directories.